cfme/tests/automate/test_service_automate.py

```diff
@@ -139,3 +141,74 @@ def test_user_requester_for_lifecycle_provision(request, appliance, provider, se
         provision_request.wait_for_request(method='ui')
         request.addfinalizer(provision_request.remove_request)
         assert result.validate(wait="60s")
+
+
+@pytest.fixture(scope="module")
+def setup_dynamic_dialog(appliance, custom_instance):
+    # Create custom instance with ruby method
+    code = dedent(
+        """
+        $evm.log(:info, "Hello World")
+        """
+    )
+    instance = custom_instance(ruby_code=code)
+
+    # Create dynamic dialog
+    dialog = "dialog_{}".format(fauxfactory.gen_alphanumeric())
+    ele_name = "ele_{}".format(fauxfactory.gen_alphanumeric())
+
+    element_data = {
+        "element_information": {
+            "ele_label": "ele_{}".format(fauxfactory.gen_alphanumeric()),
+            "ele_name": ele_name,
+            "ele_desc": fauxfactory.gen_alphanumeric(),
+            "dynamic_chkbox": True,
+            "choose_type": "Text Box",
+        },
+        "options": {"entry_point": instance.tree_path},
+    }
+    service_dialog = appliance.collections.service_dialogs.create(
+        label=dialog, description="my dialog"
+    )
+    tab = service_dialog.tabs.create(
+        tab_label="tab_{}".format(fauxfactory.gen_alphanumeric()), tab_desc="my tab desc"
+    )
+    box = tab.boxes.create(
+        box_label="box_{}".format(fauxfactory.gen_alphanumeric()), box_desc="my box desc"
+    )
+    box.elements.create(element_data=[element_data])
+
+    yield service_dialog
+    service_dialog.delete_if_exists()
```

Comment:
Required :This method does not exist in service dialog . Only delete is there .

Reply:
This method exists in `BaseEntity` so we can use it  even if it's not explicitly defined for Dialog.

Reply:
This ```delete_if_exists``` method comes from ```BaseEntity``` which is inherited by ```class Dialog(BaseEntity, Fillable):``` and this method deletes ```dialog``` successfully.